packages/babel-parser/src/tokenizer/types.js

```diff
@@ -184,7 +184,7 @@ export const keywords = Object.create(null, {
   const: makeKeywordProps("const"),
   while: makeKeywordProps("while", { isLoop }),
   with: makeKeywordProps("with"),
-  new: makeKeywordProps("new"),
+  new: makeKeywordProps("new", { beforeExpr, startsExpr }),
```

Comment:
I really hate this `beforeExpr` system, it makes it so easy to regress like this :(